File: benchmark/output/lua-love/gemini/gravity_jump.lua
local x = 384
local y = 550
local vy = 0.0
local on_ground = true
local speed = 200.0
local gravity = 500.0
local jump_force = -300.0

function love.load()
    love.window.setMode(800, 600)
    love.window.setTitle("Vibe Platformer")
end

function love.update(dt)
    -- Horizontal movement
    if love.keyboard.isDown("left") then
        x = x - speed * dt
    end
    if love.keyboard.isDown("right") then
        x = x + speed * dt
    end

    -- Vertical movement and gravity
    vy = vy + gravity * dt
    y = y + vy * dt

    -- Jump logic
    if love.keyboard.isDown("up") and on_ground then
        vy = jump_force
        on_ground = false
    end

    -- Ground collision
    if y >= 550 then
        y = 550
        vy = 0
        on_ground = true
    end
end

function love.draw()
    love.graphics.rectangle("fill", x, y, 32, 32)
    
    -- Draw ground line for reference
    love.graphics.line(0, 582, 800, 582)
end

File: vibe-lang/benchmark/output/python-pygame/gemini/gravity_jump.py
import pygame
import sys

pygame.init()
screen = pygame.display.set_mode((800, 600))
pygame.display.set_caption("Vibe Jump")
clock = pygame.time.Clock()

# Game state variables
x, y = 384.0, 550.0
vy = 0.0
on_ground = True
speed = 200.0
gravity = 500.0
jump_force = -300.0

running = True
while running:
    dt = clock.tick(60) / 1000.0

    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False
        # Handle jump on key down event
        if event.type == pygame.KEYDOWN:
            if event.key == pygame.K_UP and on_ground:
                vy = jump_force
                on_ground = False

    # Horizontal movement
    keys = pygame.key.get_pressed()
    if keys[pygame.K_LEFT]:
        x -= speed * dt
    if keys[pygame.K_RIGHT]:
        x += speed * dt

    # Vertical movement and gravity
    vy += gravity * dt
    y += vy * dt

    # Ground collision
    if y >= 550:
        y = 550
        vy = 0
        on_ground = True

    # Draw logic
    screen.fill((0, 0, 0))
    pygame.draw.rect(screen, (255, 255, 255), (int(x), int(y), 32, 32))
    
    # Draw ground line for reference
    pygame.draw.line(screen, (255, 255, 255), (0, 582), (800, 582), 2)
    
    pygame.display.flip()

pygame.quit()
sys.exit()